pkg/controller/grafanadashboard/dashboard_controller.go

```diff
@@ -150,7 +154,13 @@ func (r *ReconcileGrafanaDashboard) checkPrerequisites(d *i8ly.GrafanaDashboard)
 	}
 	d.Status.LastConfig = hash
 
-	valid, err := r.isJsonValid(d)
+	json := r.loadDashboardFromURL(d)
+	if json != "" {
+		d.Spec.Json = json
```

Comment:
Wouldn't this still be overwriting the `json` field?

Reply:
Yes, but only if the downloaded json is valid. The only other way I can see of not overwriting the `json` is changing the function `UpdateDashboard`  in the `kubeHelper.go`

Reply:
That's ok, maybe we can pass in the json instead of the whole dashboard to this function.